src/client.ts

```diff
@@ -1466,8 +1470,7 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     const defaultOptions: ChannelOptions = { state: true, watch: true, presence: false };
 
     // Make sure we wait for the connect promise if there is a pending one
-    await this.setUserPromise;
-
+    await this.wsPromise;
```

Comment:
Wait for actual wsPromise instead of setUserPromise which gets created only during connectUser. But in case of reconnection flow or manual `openConnection` ... setUserPromise would already have resolved resulting in queryChannels call without ws connection.